### PR TITLE
Fix parsing of sets in .dd files

### PR DESCRIPTION
**Pull request by samwebster**
- Fix dd_to_csv for all demos by correcting parsing for set elements with text
- Add check that column counts match when parsing dd files
- Fix errors in mapping file highlighted by check
- Allow for excel columns to be duplicated into multiple times columns
- Remove duplicate produce_times_tables

Causes small regression of ~1% on TIMS

### CODE REVIEW

File: times_mapping.txt [hunk 1]
```diff
@@ -42,7 +42,7 @@ FLO_FUNC[REG,DATAYEAR,PRC,COM_GRP,COM,TSLVL,VALUE] = ~TODO(Region,Year,TechName,
 FLO_FUNCX[REG,DATAYEAR,PRC,COM,COM,VALUE] = ~TODO(Region,Year,TechName,CommName,CommName,TODO-FLO_FUNCX)
 FLO_MARK[REG,DATAYEAR,PRC,COM,BD,VALUE] = ~FI_T(Region,Year,TechName,Comm-OUT,LimType,Attribute:FLO_MARK)
 FLO_SHAR[REG,DATAYEAR,PRC,COM,COM_GRP,ALL_TS,BD,VALUE] = ~FI_T(Region,Year,TechName,Comm-IN,Other_Indexes,TimeSlice,LimType,Attribute:FLO_SHAR)
-FLO_SUB[REG,DATAYEAR,PRC,TSLVL,CUR,VALUE] = ~TFM_INS(Region,Year,TechName,TimeSlice,Curr,Attribute:FLO_SUB)
+FLO_SUB[REG,DATAYEAR,PRC,COM,TSLVL,CUR,VALUE] = ~TFM_INS(Region,Year,TechName,Comm-IN,TimeSlice,Curr,Attribute:FLO_SUB)
```
Comment (olejandro) [inline times_mapping.txt:45]: Should it be CommName as the one below?

**Comment by olejandro** [comment]
Great! The regression on Demo001 seems to be caused by the 3 lines starting with https://github.com/samwebster/times-excel-reader/blob/sweb/fixdd/times_reader/transforms.py#L1021. Changing the first line `Dummy import of NRG commodities` to `Dummy Import of NRG` and similiarly the other two would remove the regression.

File: times_mapping.txt (hunk 1 above)
Comment (samwebster) [reply times_mapping.txt:45]: According to merged_tables.txt, ~TFM_INS has the following columns: TimeSlice,LimType,Attribute,Year,PSet_Set,PSet_PN,PSet_PD,PSet_CI,PSet_CO,CSet_Set,CSet_CN,CSet_CD,Region,VALUE,SOW,CommGrp,Stage,AllRegions,Curr,Other_Indexes

I think both are wrong and for TIMS, neither table gets generated because none of TechName, Comm-IN and CommName exist

**Comment by samwebster** [comment]
> Great! The regression on Demo001 seems to be caused by the 3 lines starting with https://github.com/samwebster/times-excel-reader/blob/sweb/fixdd/times_reader/transforms.py#L1021. Changing the first line `Dummy import of NRG commodities` to `Dummy Import of NRG` and similiarly the other two would remove the regression.

Fixes

File: times_reader/transforms.py [hunk 2]
```diff
@@ -448,9 +448,6 @@ def process_user_constraint_table(
         # Fill in UC_N blank cells with value from above
         df["UC_N"] = df["UC_N"].ffill()
 
-        if "UC_Desc" in df.columns:
-            df.drop("UC_Desc", axis=1, inplace=True)
-
```
Comment (siddharth-krishna) [inline times_reader/transforms.py:453]: Why was this removed, btw?

File: times_reader/main.py [hunk 3]
```diff
@@ -50,26 +50,23 @@ def read_mappings(filename: str) -> List[datatypes.TimesXlMap]:
             if xl_cols[-1].startswith("Attribute:"):
                 xl_cols[-1] = xl_cols[-1].replace("Attribute:", "")
                 filter_rows = True
-            col_map = {}
-            for index, value in enumerate(xl_cols):
-                col_map[value] = times_cols[index]
-            # Uppercase and validate tags:
```
Comment (siddharth-krishna) [inline times_reader/main.py:56]: Perhaps we can keep this comment

File: times_reader/main.py [hunk 4]
```diff
@@ -292,11 +289,12 @@ def produce_times_tables(
                     f" - {', '.join(missing)}"
                 )
             else:
-                cols_to_drop = [x for x in df.columns if not x in mapping.xl_cols]
+                for (times_col, xl_col) in mapping.col_map.items():
+                    df[times_col] = df[xl_col]
```
Comment (siddharth-krishna) [inline times_reader/main.py:293]: Perhaps a comment here explaining that this allows excel cols to be duplicated into multiple times columns?

**Review by siddharth-krishna — APPROVED** [review]

File: times_reader/transforms.py (hunk 2 above)
Comment (samwebster) [reply times_reader/transforms.py:453]: Because we now need that column for the UC_N mapping:

File: times_reader/main.py (hunk 4 above)
Comment (samwebster) [reply times_reader/main.py:295]: Done

File: times_reader/main.py (hunk 3 above)
Comment (samwebster) [reply times_reader/main.py:56]: Done

File: times_mapping.txt (hunk 1 above)
Comment (samwebster) [reply times_mapping.txt:45]: I think these should be Comm-IN
